tests/allocator-test.cc

```diff
@@ -87,6 +87,65 @@ namespace
 		  "Freed pointer still live; load barrier or revoker out of service?");
 	}
 
+	__noinline void test_heap_free()
+	{
+		debug_log("Testing heap_free edge cases");
+		Timeout          t{5};
+		Capability<void> p1 = heap_allocate(&t, MALLOC_CAPABILITY, 16);
+		TEST(p1.is_valid(), "Unable to make allocation for interior free test");
+
+		// Attempt to free with wrong heap capability
```

Comment:
How had we not tested that already?

Reply:
You mean the interior pointer free? It's possible we already do -- there are a lot of allocator tests! Actually I just noticed one of the tests I added is a duplicate (free with wrong quota) -- will remove that.